### PR TITLE
Extending snapshot mechanism to allow saving and restoring of OS and Loader information

### PR DESCRIPTION
**Pull request by liba2k**
Handling mainly UEFI Loader and OS, For for other OSs will have to be added as needed.

### CODE REVIEW

File: qiling/core.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def enable_lib_patch(self):
 
 
     # save all qiling instance states
-    def save(self, reg=True, mem=True, fd=False, cpu_context=False, snapshot=None):
+    def save(self, reg=True, mem=True, fd=False, cpu_context=False, os_context=True, loader=True, snapshot=None):
```
Comment (xwings) [inline qiling/core.py:241]: should os_context and loader = False ?

**Comment by xwings** [comment]
btw, update changelog too :)

File: qiling/core.py (hunk 1 above)
Comment (liba2k) [reply qiling/core.py:241]: Whatever you think will serve the project better.

**Comment by xwings** [comment]
Cool, thanks. Merged.

**Pull request merged**